### PR TITLE
fix(l10n): Rephrasing some exceptions to be translated

### LINKED ISSUE #47773: chore(l10n): Bad english needs to be improved
https://github.com/nextcloud/server/blob/332b3efdf0df6bd489d83f58e8ffa677749d0774/lib/private/Share20/Manager.php#L133

Reported at Transifex.

Should "SharedWith" a variable?

@solracsf Please help and improve

### PR DESCRIPTION
**Pull request by solracsf**
* Resolves: https://github.com/nextcloud/server/issues/47773

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -130,34 +130,34 @@ protected function generalCreateChecks(IShare $share, bool $isUpdate = false) {
 		if ($share->getShareType() === IShare::TYPE_USER) {
 			// We expect a valid user as sharedWith for user shares
 			if (!$this->userManager->userExists($share->getSharedWith())) {
-				throw new \InvalidArgumentException($this->l->t('SharedWith is not a valid user'));
+				throw new \InvalidArgumentException($this->l->t('Share recipient is not a valid user'));
```
Comment (susnux) [inline lib/private/Share20/Manager.php:133]: I still think this should have been never be translated.

**Review by susnux — APPROVED** [review]
New messages seem to be fine.

Still I think we should not translate those errors, as this are not user facing exception (no HintException). Meaning those are developers only and translating them makes debugging log files much harder.

**Comment by solracsf** [comment]
In the other side, if the user understands the Exception, he may be able to solve the situation without opening an issue... A coin has 2 sides 🫣

**Review by provokateurin — APPROVED** [review]
I agree with @susnux, all non user facing error messages should not be translated. The fix is good though.

**Comment by provokateurin** [comment]
> In the other side, if the user understands the Exception, he may be able to solve the situation without opening an issue... A coin has 2 sides 🫣

They can use a translator for that. The harder part is if we get a translated error message and have to figure out what the original message was. A translator might not give the exact original version and it is much harder to find the source (unless you got a stacktrace).

**Comment by susnux** [comment]
> In the other side, if the user understands the Exception, he may be able to solve the situation without opening an issue... 

But the user will not see this, or if they see it they can not fix it because those errors happen only because of invalid API usage -> broken apps or server -> if the user can fix it they are developers and probably should also understand English in our code base. On the otherhand as @provokateurin mentioned for us this makes reading issues / debugging much harder as we need to guess which error message was meant.

**Review by rakekniven — APPROVED** [review]

**Comment by joshtrichards** [comment]
Motivated by translation requirements or not (and whether for users or developers), I quite like the added clarity of these rephrasings. Good stuff!

**Comment by susnux** [comment]
@solracsf you need to adjust the tests, they expect some strings that now have been changed

**Comment by joshtrichards** [comment]
Tests are failing:

```
There were 2 failures:

1) Test\Share20\ManagerTest::testMoveShareUserNotRecipient
Failed asserting that exception message 'Invalid share recipient' contains 'Invalid recipient'.

2) Test\Share20\ManagerTest::testMoveShareGroupNotRecipient
Failed asserting that exception message 'Invalid share recipient' contains 'Invalid recipient'.

```

**Pull request merged**